package/MDAnalysis/core/groups.py

```diff
@@ -850,6 +850,128 @@ def center_of_geometry(self, pbc=None, compound='group'):
         return self.center(None, pbc=pbc, compound=compound)
 
     centroid = center_of_geometry
+    
+    @warn_if_not_unique
+    def accumulate(self, attribute, op=np.sum, compound='group'):
+        """Accumulates the attribute associated with (compounds of) the group.
+        
+        Accumulates the attribute of :class:`Atoms<Atom>` in the group.
+        The accumulation is done per :class:`Residue`, :class:`Segment`, molecule, 
+        or fragment can be obtained by setting the `compound` parameter
+        accordingly. By default the method sums up all elements, but 
+        any numpy operation applying to numpy.ndarray can be used.
+
+        Parameters
+        ----------
+        attribute : str
```

Comment:
I liked that you could give an arbitrary array (assuming it has the right size). In the ideal case, I would like to be able to provide either the name, or an arbitrary array.

Reply:
Yes. This should be feasible.